### PR TITLE
Check skillmap user locale against target's available locales before setting

### PR DESCRIPTION
**Pull request by shakao**
we were always passing the user's locale to crowdin, even if it wasn't in the list of available locales for that target. this checks if there's a `force` flag or if the locale has been enabled, and if not it resets to the target's default locale. 

i tested `forcelang=ru`, as well as setting the browser locale to an enabled (chinese) and non-enabled (russian) locale and it seems to be working as expected (localizes for the first two cases, doesn't for the third), but let me know if there's any scenarios i missed!

### CODE REVIEW

**Review by dmonroym — APPROVED** [review]

File: skillmap/src/App.tsx [hunk 1]
```diff
@@ -164,9 +168,7 @@ class AppImpl extends React.Component<AppProps, AppState> {
             force: force,
         });
 
-        if (pxt.Util.isLocaleEnabled(useLang!)) {
-            pxt.BrowserUtils.setCookieLang(useLang!);
-        }
+        pxt.BrowserUtils.setCookieLang(useLang!);
```
Comment (abchatra) [inline skillmap/src/App.tsx:171]: We set he cookie to userLang only if isLocaleEnabled right? Why this change?

File: skillmap/src/App.tsx [hunk 2]
```diff
@@ -138,13 +138,17 @@ class AppImpl extends React.Component<AppProps, AppState> {
             force = !!mlang && !!mlang[2];
         }
 
+        const defLocale = pxt.appTarget.appTheme.defaultLocale;
+        if (!force && !pxt.Util.isLocaleEnabled(useLang!)) {
+            useLang = defLocale;
```
Comment (abchatra) [inline skillmap/src/App.tsx:143]: I don't get this one as well. You are setting useLang if it is not isLocaleEnabled?

Comment (abchatra) [reply skillmap/src/App.tsx:143]: Sorry ! tripped me off. This looks good.

File: skillmap/src/App.tsx (hunk 1 above)
Comment (shakao) [reply skillmap/src/App.tsx:171]: oh it's being covered by the above case (invalid languages are overwritten by the default), by the time it gets here the useLang should only be an enabled locale! i can add it back though, it doesn't hurt either way

Comment (abchatra) [reply skillmap/src/App.tsx:171]: But when forcelang is used its not overridden right?

Comment (shakao) [reply skillmap/src/App.tsx:171]: OH yes you're totally right, i'll replace the check! hmm @jwunderl do we want to set the user language when the forcelang flag is used? or should we also be skipping that?

Comment (jwunderl) [reply skillmap/src/App.tsx:171]: I would lean towards maintaining the current behavior of not persisting unapproved languages / only keeping them for the session when forcelang is used; forcelang is typically just a test scenario anyway

**Review by jwunderl — APPROVED** [review]
Besides maybe not persisting forcelang lgtm~

File: skillmap/src/App.tsx (hunk 1 above)
Comment (shakao) [reply skillmap/src/App.tsx]: sounds good! i'm moving the setUserLanguage out of `updateLocalizationAsync` and next to the cookie setting, then wrapping them both in this check. this is different from the previous skillmap behavior but i think matches the webapp?

**Review by abchatra — APPROVED** [review]

**Comment by abchatra** [comment]
port to stable
